### DISCUSSION TITLE
The comparison issue for TS

### DISCUSSION THREAD

**Discussion by HAIXIN-ZHANG** [DefinitelyTyped/DefinitelyTyped#66430] [Request a new @types package]
Hey, 
just wondering why TS does not support something like `String?.length  ==> 0` or `Array?.length  ==> 0`. As TS is already supported to use `Array?.length  == 0` or` Array?.length  === 0`. but why no` ==>, =>, ==<, =< ` .

If we want to do something like `if (String?.length >0) {}` we currently have to make it to` if (String && String.length >0) {}` or ` if (!String?.length ===0) {}`, but if we follow the Syntax logic of TS, it should have a Syntax like `if (String?.length ==>0) {}`

**Comment by Semigradsky** [comment]
You can use `String?.length ?? 0 > 0`. Looks pretty and clear for me
